packages/cli/generators/relation/base-relation.generator.js

```diff
@@ -98,23 +98,17 @@ module.exports = class BaseRelationGenerator extends ArtifactGenerator {
       type: this._getRepositoryRelationPropertyType(),
     };
 
-    if (relationUtils.doesPropertyExist(classDeclaration, property.name)) {
-      throw new Error(
-        'property ' + property.name + ' already exist in the repository.',
-      );
-    } else {
-      relationUtils.addProperty(classDeclaration, property);
-    }
+    // already chcked the existence of property before
+    relationUtils.addProperty(classDeclaration, property);
   }
 
   _addParametersToRepositoryConstructor(classConstructor) {
     const parameterName =
       utils.camelCase(this.artifactInfo.dstRepositoryClassName) + 'Getter';
 
     if (relationUtils.doesParameterExist(classConstructor, parameterName)) {
-      throw new Error(
-        'Parameter ' + parameterName + ' already exist in the constructor.',
-      );
+      // no need to check if the getter already exists
```

Comment:
Doesn't parameter exist imply the relation already exists? And therefore an error should be thrown like the deleted code?

Reply:
The param is getter, such as `@repository.getter('OrderRepository') protected orderRepositoryGetter: Getter<OrderRepository>`. And I don't think this implies a certain _relation name_ already exists. That's why I removed it. 

And I was also thinking about some edge cases. For example, since we allow customized `keyFrom`, user could have two _same_ relations with 2 models with two different names. In this case, they will share the same getter.

Reply:
> since we allow customized keyFrom, user could have two same relations with 2 models with two different names. In this case, they will share the same getter.

👍 Thank you I agree with you.
The property should be checked instead of parameter.